src/scripts/choices.js

```diff
@@ -812,13 +812,13 @@ class Choices {
     this.input.focus();
   }
 
-  _handleChoiceAction(activeItems, element) {
-    if (!activeItems || !element) {
+  _handleChoiceAction(activeItems, element, idDirectly) {
```

Comment:
I'm confused - I can't see `idDirectly` ever being passed to this function? It's name is also not particularly clear

Reply:
@jshjohnson : it appeared to make sense in our use case of a 'Tags' field: in the formio.js library, in a multiple-value Select component with the Choice options being other formio.js Resources, the requirement was to allow easy creation of a new Resource ('Tag').  We implemented it using this new callback, with Shift + Enter being the combination to force creating a new tag unconditionally.  Here's the full callback source code which apparently works as required:
```
    // The following function gets 'this' from Choices ...
    var _this = this;
    const callbackOnUnknownChoice = function(value, hasShift) {
      var choices = this;
      const reject = function() {
        choices.dropdown.isActive = false;
        choices.showDropdown();
        choices.input.focus();
        choices.clearInput();
        return false;
      };
      if (this._store.activeItems.findIndex(x => (x.value.data.text === value)) > -1) {
        alert(`Tag "${value}" already selected!`);
        return reject();
      }
      var idx = this._store.choices.findIndex(x => (x.value.data.text === value));
      if (idx > -1) {
        const item = this._store.choices[idx];
        console.log(`Tag "${value}" already exists: `,item);
        this.highlightItem(item);
        const activeItems = this._store.activeItems;
        if (activeItems[0]) {
          activeItems[0].keyCode = 13;
        }
        this._handleChoiceAction(this._store.activeItems, null, item.id);
        return false;
      }
      if (!hasShift) {
        if (!confirm(`New Tag "${value}" ?`)) {
          return reject();
        }
      }
      const dialog = _this.createModal(_this.component.addResourceLabel || 'Add Resource');
      const formioForm = _this.ce('div');
      dialog.body.appendChild(formioForm);
      const form = new Webform(formioForm);
      form.on('render', () => {
        form.submit();
      });
      form.on('submit', (submission) => {
        console.log(`Created new tag "${value}"`);
        const newTagObj = submission;
        if (_this.component.multiple) {
          submission = [..._this.dataValue, submission];
        }
        form.destroy();
        dialog.close();
        //
        const newItem = {
          label: value,
          value: newTagObj,
          keyCode: 13,
          isSelected: true,
          active: false
        };
        //console.log('Adding choice ',newItem);
        this.setChoices([newItem],'value','label',false);
        const idx = this._store.choices.findIndex(x => (x.value.data.text === value));
        if (idx > -1) {
          const item = this._store.choices[idx];
          //console.log('Activating item ',item.id);
          this.highlightItem(item);
          const activeItems = this._store.activeItems;
          if (activeItems[0]) {
            activeItems[0].keyCode = 13;
          }
          this._handleChoiceAction(this._store.activeItems, null, item.id);
        }
        else {
          console.warn('added choice not found: ',value);
        }
        this.clearInput();
      });
      form.submission = { data: { text: value } };
      form.src = `${_.get(this.root, 'formio.projectUrl', Formio.getBaseUrl())}/form/${_this.component.resource}`;
      return true;
    };
```
As you can see, `idDirectly` is used there. If you have an idea for a better name, or for a better way to expose that API, feel free to change as you like.